File: makeup-app/src/ChildLabor.py
import requests
import pandas as pd
import os
from dotenv import load_dotenv

load_dotenv()

def get_working_percent(country_name):
    api_key = os.getenv('DOL_API_KEY')
    
    if not api_key:
        print("Error: DOL_API_KEY not found in .env file.")
        return None
    
    base_url = "https://apiprod.dol.gov/v4/get/ILAB/LaborShield_ReportingData/json"
    
    params = {
        "limit": "1",
        "offset": "0",
        "fields": "working_percent",
        "filter_object": f'{{"field": "country", "operator": "eq", "value": "{country_name}"}}',
        "X-API-KEY": api_key
    }

    try:
        response = requests.get(base_url, params=params)
        response.raise_for_status() 
        
        raw_data = response.json()

        data_json = raw_data.get('data', [])
        
        df = pd.DataFrame(data_json)
        df_result = df['working_percent'].iloc[0]
        return float(str(df_result).replace('%', ''))

    except Exception as e:
        print(f"Error occurred: {e}")
        return None
    
def calculate_score(n):
    # calculate so anything higher tham 50% is a 0, after that each percent removes 2 from the score out of 100
    if (n >= 50):
        return 0
    return int(100 - (n * 2))

def countries_to_score(countries):
    description = ""
    country_list = [countries[0], countries[1]]
    country_scores = []
    final_score = -1

    for country in country_list:
        new_val = get_working_percent(country)
        if (new_val != None): country_scores.append(new_val)
    
    if len(country_scores) == 0:
        description += "Unfortunately no information was found on these countries"
    else:
        final_score = int(sum(country_scores) / len(country_scores))
        description += "This score was created by evaluating the child labor practices of " + countries[0] + " and " + countries[1]
        description += " using Department of Labor statistics."

    report = {
        "labor_score": final_score,
        "labor_description": description
    }
    
    return report

#result2 = countries_to_score(['Chad', 'Somalia'])
#print(result2)

File: makeup-app/src/childLabor.ts
const DOL_API_KEY = import.meta.env.VITE_DOL_API_KEY

interface LaborReport {
  score: number | "unknown"
  reason: string
}

async function getWorkingPercent(countryName: string): Promise<number | null> {
  if (!DOL_API_KEY) {
    console.error("VITE_DOL_API_KEY not found in .env file.")
    return null
  }

  const params = new URLSearchParams({
    limit: "1",
    offset: "0",
    fields: "working_percent",
    filter_object: JSON.stringify({ field: "country", operator: "eq", value: countryName }),
    "X-API-KEY": DOL_API_KEY,
  })

  const url = `https://apiprod.dol.gov/v4/get/ILAB/LaborShield_ReportingData/json?${params}`

  try {
    const result = await new Promise<{ success: boolean; data?: { data?: { working_percent: string }[] }; error?: string }>((resolve, reject) => {
      chrome.runtime.sendMessage({ type: 'FETCH_DOL', url }, (response) => {
        if (chrome.runtime.lastError) {
          reject(new Error(chrome.runtime.lastError.message))
        } else {
          resolve(response)
        }
      })
    })

    if (!result.success) throw new Error(result.error)

    const data = result.data?.data ?? []
    if (data.length === 0) return null

    const value = String(data[0].working_percent).replace("%", "")
    return parseFloat(value)
  } catch (e) {
    console.error(`Error fetching labor data for ${countryName}:`, e)
    return null
  }
}

export async function countriesToScore(countries: string[]): Promise<LaborReport> {
  const countryList = countries.slice(0, 2)
  const countryScores: number[] = []

  for (const country of countryList) {
    const val = await getWorkingPercent(country)
    if (val != null) countryScores.push(val)
  }

  if (countryScores.length === 0) {
    return {
      score: "unknown",
      reason: "Unfortunately no information was found on these countries",
    }
  }

  const avg = countryScores.reduce((a, b) => a + b, 0) / countryScores.length
  // anything >= 50% child labor → score 0; each percent removes 2 points from 100
  const score = avg >= 50 ? 0 : Math.round(100 - avg * 2)

  return {
    score,
    reason:
      `This score was created by evaluating the child labor practices of ${countries[0]} and ${countries[1]}` +
      ` using Department of Labor statistics.`,
  }
}
